app/controllers/books_controller.rb

```diff
@@ -0,0 +1,31 @@
+class BooksController < ApplicationController
+  before_action :set_position, only: [:show, :index]
+
+  def show
+    @book = Book.find(params[:id])
+    if current_user
+      @review = Review.new(book_id: @book.id, user_id: current_user.id, title: current_user.name)
+    end
+  end
+
+  def index
```

Comment:
You can introduce a presenter object to get rid of this massive amount of instance variables
Take a look here:
https://thepugautomatic.com/2014/03/draper/

If you are not sure how to make it - just let me know, I will help you